### DISCUSSION TITLE
examples for session_update

### DISCUSSION THREAD

**Discussion by sirpy** [WalletConnect/walletconnect-monorepo#3399] [Developer Support]
Can you please add some concrete examples in the docs or here how session_update is supposed to be used?
How does a wallet trigger it? with what content? how does the dapp should handle it?
How does a dapp trigger it? with what content? how does the wallet handle it?

For example in the tests i've noticed a client can request wallet_addChain and that actually triggers a session_update
In the docs it shows something about adding an account and a chain, but not clear if it is the wallet? the dapp? how should it be handled

**Comment by glitch-txs** [comment]
we'll work on this, thanks for the feedback 🙏
